runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -2463,12 +2463,19 @@ void polling_task(void* ignore)
       // not oversubscribed, the trip through sched_yield(2) if no
       // other process or pthread can run is quite quick.
       //
+      if (spins++ % 1024 == 0) {
```

Comment:
Out of curiosity, was this 1024 value chosen because it strikes the best (in some sense) balance between not adding too much overhead due to checkpoint calls and not delaying privatization deletes too long?  Or because on average it produces a certain rate of checkpoint calls per second?  Or do we just know that for a fast work-waiting loop like this we don't want to do a checkpoint call anywhere near to every time around?

Reply:
Unfortunately as I was unable to perform appropriate profiling for when the task is polling without work (`GNI_RC_NOT_DONE`), but since each checkpoint takes on average 30 nanoseconds to run, I believe that the checkpoint does not need to be invoked on every loop iteration. The value '1024' was chosen arbitrarily however, but it should be sufficient to allow processing for rapid concurrent PUT/GET operations. As well I have the number of spins reset after each successful PUT/GET operation to ensure that we do not invoke a checkpoint when we're super busy.

Reply:
That doesn't really answer what I was asking.  Let me try it from a different direction.  Would the only effect of doing `chpl_qsbr_checkpoint()` calls less frequently here be to delay when QSBR-managed memory that was eligible for deletion actually got deleted?  That is, if I were to change this code such that the calls were made, say, every 0.1 second instead of every N times around the loop, would the only effect be that some QSBR-managed memory would get deleted 0.1 second later that it was actually eligible to be deleted?

Reply:
Yep, if you call `chpl_qsbr_checkpoint` once every N seconds, then memory can only be reclaimed at most once every N seconds.

Reply:
Okay, great, thanks!